bin/propolis-server/src/lib/spec/builder.rs

```diff
@@ -113,18 +116,51 @@ impl SpecBuilder {
         Ok(())
     }
 
+    /// Sets the spec's boot order to the list of disk devices specified in
+    /// `boot_options`.
+    ///
+    /// All of the items in the supplied `boot_options` must already be present
+    /// in the spec's disk map.
+    pub fn add_boot_order(
+        &mut self,
+        component_name: String,
+        boot_options: impl Iterator<Item = BootOrderEntry>,
+    ) -> Result<(), SpecBuilderError> {
+        if self.component_names.contains(&component_name) {
+            return Err(SpecBuilderError::ComponentNameInUse(component_name));
+        }
+
+        if self.spec.boot_settings.is_some() {
+            return Err(SpecBuilderError::BootSettingsInUse);
+        }
+
+        self.spec.boot_settings =
+            Some(BootSettings { name: component_name, order: vec![] });
+
+        for item in boot_options {
+            self.add_boot_option(item)?;
+        }
+
+        Ok(())
+    }
+
     /// Add a boot option to the boot option list of the spec under construction.
-    pub fn add_boot_option(
+    fn add_boot_option(
```

Comment:
fwiw i think this is of substantially less utility since you must have called `add_boot_order()` with at least an empty list for this to not panic on accessing `boot_settings`. as-is, most of this function, by vertical space, is just dancing around the fact that the caller just set `self.spec.boot_settings` to `Some`.

lets fold this into `add_boot_order` and be done with it?